Source/Core/Core/MemoryWatcher.h

```diff
@@ -0,0 +1,40 @@
+// Copyright 2015 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <atomic>
+#include <map>
+#include <sys/socket.h>
+#include <sys/un.h>
+
+#include "Common/Thread.h"
```

Comment:
Thread shouldn't be included here, since nothing in the header requires it.

You can just include `<thread>`